### PR TITLE
tests/provider: Update resource testing to 0.12 syntax (IAM Resources)

### PR DESCRIPTION
**Pull request by YakDriver**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Reference: https://github.com/terraform-providers/terraform-provider-aws/issues/8950

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
NONE
```

Output from acceptance testing: Handled via daily acceptance testing

### CODE REVIEW

File: aws/resource_aws_iam_user_login_profile_test.go [hunk 1]
```diff
@@ -354,12 +354,14 @@ func testAccAWSUserLoginProfileConfig_PasswordLength(rName, path, pgpKey string,
 %s
 
 resource "aws_iam_user_login_profile" "user" {
-  user            = "${aws_iam_user.user.name}"
+  user            = aws_iam_user.user.name
   password_length = %d
 
   pgp_key = <<EOF
 %s
+
```
Comment (bflad) [inline aws/resource_aws_iam_user_login_profile_test.go:362]: This extra newline is causing the tests to fail:

```
--- FAIL: TestAccAWSUserLoginProfile_basic (22.05s)
```

```suggestion
```

File: aws/resource_aws_iam_user_login_profile_test.go [hunk 2]
```diff
@@ -369,11 +371,13 @@ func testAccAWSUserLoginProfileConfig_Required(rName, path, pgpKey string) strin
 %s
 
 resource "aws_iam_user_login_profile" "user" {
-  user = "${aws_iam_user.user.name}"
+  user = aws_iam_user.user.name
 
   pgp_key = <<EOF
 %s
+
```
Comment (bflad) [inline aws/resource_aws_iam_user_login_profile_test.go:378]: Same here

```suggestion
```

Comment (YakDriver) [reply aws/resource_aws_iam_user_login_profile_test.go]: :blargh:

**Comment by YakDriver** [comment]
```
--- PASS: TestAccAWSUserLoginProfile_basic (22.02s)
```

**Pull request merged**